simple/sources/simple_stmt.c

```diff
@@ -599,7 +599,7 @@ int simple_parser_stmt ( Parser *parser )
 				simple_parser_icg_newoperation(parser,ICO_LOADAPUSHV);
 				simple_parser_icg_newoperand(parser,cStr);
 				simple_parser_icg_newoperation(parser,ICO_LOADBLOCK);
-				simple_parser_icg_newoperand(parser,"lengthOf");
+				simple_parser_icg_newoperand(parser,"length_of_minus_one");
```

Comment:
What's with the new block ?

Reply:
The `lengthOf` block returns the actual length of the list e.g the following list 
```js
list = [1,2,3,4,5,6]
@lengthOf(list)
```
Output :
```js
6
```
Since simple index starts from 0 and not one then list[6] is definitely an error so the last index is always the length of the list minus one. The `for...in` loop does uses the `lengthOf` to determine the size of the list but will result in error trying to get the last index, so there is need for another block `length_of_minus_one` which does return the `lengthOf` list minus 1. The block is not meant to use outside the compiler

Reply:
Ohh... Great then, this block resolves looping index issues. :+1: